types/pllua/pllua/elog.d.tl

```diff
@@ -0,0 +1,65 @@
+local record elog
+   enum Severity
+      "debug"
+      "log"
+      "info"
+      "notice"
+      "warning"
+      "error"
+   end
+
+   record ELogInfo
+      sqlstate: string
+      message: string
+      detail: string
+      table: string
+      column: string
+      datatype: string
+      constraint: string
+      schema: string
+   end
+
+   elog: function(Severity, message: string)
+   elog: function(Severity, sqlstate: string, message: string)
+   elog: function(Severity, sqlstate: string, message: string, detail: string)
+   elog: function(Severity, sqlstate: string, message: string, detail: string, hint: string)
+   elog: function(Severity, ELogInfo)
```

Comment:
Is there any way to "re-export" a module or a bunch of symbols?  At least being able to do something like:
```
local elog = require 'pllua.elog'
[...]
   elog: typeof(elog.elog)
   debug: typeof(elog.debug)
   log: typeof(elog.log)
   info: typeof(elog.info)
   notice: typeof(elog.notice)
   warning: typeof(elog.warning)
   error: typeof(elog.error)
```

Something like typeof would also be helpful for referencing local types that could not be exported.

Reply:
The nested type should be publicly accessible just like the record attributes, so you should be able to use `elog.ELogInfo` from client code that requires this module.

Reply:
Yes, `elog.ELogInfo` is being exported (and it's being used in `spi.d.tl`), however I meant being able to express a "poly" type and associate that to a type variable, something like `type logfunction = function(message: string) & function(sqlstate: string, message: string) & ...` but I guess we first would need some syntax to express type intersection.